aten/src/ATen/core/ivalue_inl.h

```diff
@@ -684,6 +684,13 @@ inline std::vector<int64_t> IValue::toIntVector() const {
   AT_ASSERT(isIntList(), "Expected IntList but got ", tagKind());
   return createVectorFromList<int64_t>(static_cast<const c10::detail::ListImpl*>(payload.as_intrusive_ptr));
 }
+// handles conversion from std::vector<int64_t> to at::IntArrayRef within optional
+inline c10::optional<at::IntArrayRef> IValue::toIntVectorOptional() {
+  if (isNone()) {
+    return c10::nullopt;
+  }
+  return at::IntArrayRef(toIntVector());
```

Comment:
This isn't safe. The vector will be destroyed as this function returns, and we'll return an IntArrayRef that points to freed memory.

Reply:
Yup, haven't updated the PR description but I'm working through these obvious ones now.